client/ayon_core/plugins/publish/extract_color_transcode.py

```diff
@@ -279,15 +295,13 @@ def _translate_to_sequence(self, files_to_convert):
                     "Too many collections {}".format(collections))
 
             collection = collections[0]
-            frames = list(collection.indexes)
-            if collection.holes():
-                return files_to_convert
-
-            frame_str = "{}-{}#".format(frames[0], frames[-1])
-            file_name = "{}{}{}".format(collection.head, frame_str,
-                                        collection.tail)
 
-            files_to_convert = [file_name]
+            # Support sequences with holes by supplying dedicated `--frames`
+            # Create `frames` string like "1001-1002,1004,1010-1012
+            frames: str = collection.format("{ranges}").replace(" ", "")
+            # Create `filename` string like "file.%04d.exr"
+            filename = collection.format("{head}{padding}{tail}")
+            return [(filename, frames)]
```

Comment:
@iLLiCiTiT any quick ideas on how to improve this function to have a more consistent return type. :D My brain is tired today it seems.

Reply:
Changed it around - not sure if I'm entirely happy cosmetically - but it should work fine.